checkpoint-single-nic/azuredeploy.json

```diff
@@ -0,0 +1,305 @@
+{
+    "$schema": "http://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json",
+    "contentVersion": "1.0.0.0",
+    "parameters": {
+        "adminPassword": {
+            "metadata": {
+                "description": "Administrator password"
+            },
+            "type": "securestring"
+        },
+        "authenticationType": {
+            "allowedValues": [
+                "password",
+                "sshPublicKey"
+            ],
+            "defaultValue": "password",
+            "metadata": {
+                "description": "Authentication type"
+            },
+            "type": "string"
+        },
+        "newStorageAccountName": {
+            "metadata": {
+                "description": "Name of storage account"
+            },
+            "type": "string"
+        },
+        "offer": {
+            "allowedValues": [
+                "R7730BYOL",
+                "R7730PAYG"
+            ],
+            "defaultValue": "R7730BYOL",
+            "metadata": {
+                "description": "Offer version and type"
+            },
+            "type": "string"
+        },
+        "sshPublicKey": {
+            "defaultValue": "",
+            "metadata": {
+                "description": "Administrator SSH public key"
+            },
+            "type": "string"
+        },
+        "vmName": {
+            "metadata": {
+                "description": "Name of the Security Gateway"
+            },
+            "type": "string"
+        },
+        "vmSize": {
+            "defaultValue": "Standard_D3",
+            "metadata": {
+                "description": "Size of the VM"
+            },
+            "type": "string"
+        }
+    },
+    "variables": {
+        "apiVersion": "2015-06-15",
+        "customData": "#!/bin/bash",
+        "addressPrefix": "10.0.0.0/16",
+        "gatewayPrivateAddress": "10.0.1.10",
+        "onPremiseNetwork": "198.51.100.0/24",
+        
+        "imageOffer": "check-point-r77-10",
+        "imagePublisher": "checkpoint",
+        "imageReference": "[variables(concat('imageReference',parameters('offer')))]",
+        "imageReferenceR7730BYOL": {
+            "offer": "[variables('imageOffer')]",
+            "publisher": "[variables('imagePublisher')]",
+            "sku": "sg-byol",
+            "version": "latest"
+        },
+        "imageReferenceR7730PAYG": {
+            "offer": "[variables('imageOffer')]",
+            "publisher": "[variables('imagePublisher')]",
+            "sku": "sg-ngtp",
+            "version": "latest"
+        },
+        "nicName": "gateway-nic",
+        "osProfile": "[variables(concat('osprofile',parameters('authenticationType')))]",
+        "osProfilepassword": {
+            "adminPassword": "[parameters('adminPassword')]",
+            "adminUsername": "notused",
```

Comment:
The admin user name is always admin. 
You need to pass a user name in the profile but our image always disregards it and sets a user called admin.